perception/sub8_perception/src/sub8_perception/start_gate.cc

```diff
@@ -0,0 +1,191 @@
+#include <sub8_perception/start_gate.hpp>
+Sub8StartGateDetector::Sub8StartGateDetector() : image_transport(nh), gate_line_buffer(30), running(true) {
+    image_sub = image_transport.subscribeCamera("stereo/right/image_raw", 1,
+                &Sub8StartGateDetector::image_callback, this);
+    service_2d = nh.advertiseService("/vision/start_gate/pose", &Sub8StartGateDetector::request_start_gate_position_2d, this);
+    service_enable = nh.advertiseService("/vision/start_gate/enable", &Sub8StartGateDetector::request_start_gate_enable, this);
+}
+Sub8StartGateDetector::~Sub8StartGateDetector() {}
+
+void Sub8StartGateDetector::image_callback(const sensor_msgs::ImageConstPtr &image_msg,
+        const sensor_msgs::CameraInfoConstPtr &info_msg) {
+
+    if (running) {
+
+        cam_model.fromCameraInfo(info_msg);
+        image_time = image_msg->header.stamp;
+
+        if (gate_line_buffer.size() > 29) {
+
+            for (size_t i = 0; i < gate_line_buffer.size(); i++) {
```

Comment:
Since you're not doing arithmetic with the indexing variable, you should use range-based for loops.